jina/drivers/rank/__init__.py

```diff
@@ -49,11 +48,10 @@ def _apply_all(self, docs: 'DocumentSet', context_doc: 'Document', *args,
             - Set the ``traversal_paths`` of this driver such that it traverses along the ``matches`` of the ``chunks`` at the level desired.
         """
 
-        # if at the top-level already, no need to aggregate further
-        query_meta = context_doc.get_attrs(*self.exec.required_keys)
-
+        query_meta = context_doc.get_attrs(*self.exec.required_keys) if hasattr(self.exec, 'required_keys') else None
         old_match_scores = {match.id: match.score.value for match in docs}
-        match_meta = {match.id: match.get_attrs(*self.exec.required_keys) for match in docs}
+        match_meta = {match.id: match.get_attrs(*self.exec.required_keys) for match in docs} if hasattr(self.exec, 'required_keys') else None
```

Comment:
For https://github.com/jina-ai/jina-hub/pull/3670 it is important to make sure that the `required_keys` are obtained in the exact same order as in the `required_keys`